src/lib.rs

```diff
@@ -165,57 +161,51 @@
 #[allow(clippy::single_component_path_imports)]
 use rstest_reuse;
 
-// This import is used in the doc comments and test code above. Clippy incorrectly
-// declares it an unused import.
-#[allow(unused_imports)]
-use element::Element;
-
-pub mod binary;
-pub mod result;
-// Public as a workaround for: https://github.com/amazon-ion/ion-rust/issues/484
-pub mod constants;
-pub mod data_source;
-pub mod element;
-pub(crate) mod raw_reader;
-pub mod text;
-pub mod types;
-
-mod ion_data;
-#[cfg(feature = "ion-hash")]
-pub mod ion_hash;
-
-pub(crate) mod blocking_reader;
+// Private modules that serve to organize implementation details.
+mod binary;
+mod blocking_reader;
 mod catalog;
-
+mod constants;
+mod data_source;
+mod ion_data;
 mod ion_reader;
 mod ion_writer;
+mod raw_reader;
 mod raw_symbol_token;
 mod raw_symbol_token_ref;
-pub(crate) mod reader;
+mod reader;
 mod shared_symbol_table;
 mod symbol_ref;
 mod symbol_table;
 mod system_reader;
+mod text;
+
+// Publicly-visible modules with nested items which users may choose to import
+pub mod element;
+pub mod result;
+pub mod types;
```

Comment:
So, just to make sure I'm understanding correctly, we're exporting everything here and at the root?

Reply:
> we're exporting everything here

There are some types that I don't think are commonly used enough (at least by name) to merit a spot in the crate root (`OwnedElementIterator`, `FractionalSecondSetter`, etc).

That said, there aren't _that_ many of them and they're pretty distinctively named, so I could also be convinced not to expose any modules at all.